geospaas_harvesting/ingesters.py

```diff
@@ -539,15 +540,30 @@ def _get_normalized_attributes(self, dataset_info, *args, **kwargs):
         # set optional ForeignKey metadata from Nansat or from defaults
         normalized_attributes['gcmd_location'] = n_metadata.get(
             'gcmd_location', pti.get_gcmd_location('SEA SURFACE'))
-        normalized_attributes['provider'] = n_metadata.get(
-            'data_center', pti.get_gcmd_provider('NERSC'))
+        normalized_attributes['provider'] = pti.get_gcmd_provider(
+            n_metadata.get('provider', 'NERSC'))
         normalized_attributes['iso_topic_category'] = n_metadata.get(
             'ISO_topic_category', pti.get_iso19115_topic_category('Oceans'))
 
         # Find coverage to set number of points in the geolocation
         if len(nansat_object.vrt.dataset.GetGCPs()) > 0:
             nansat_object.reproject_gcps()
         normalized_attributes['location_geometry'] = GEOSGeometry(
-            nansat_object.get_border_wkt(nPoints=n_points), srid=4326)
+            nansat_object.get_border_wkt(n_points=n_points), srid=4326)
+        json_dumped_dataset_parameters = n_metadata.get('dataset_parameters', None)
+        if json_dumped_dataset_parameters:
+            try:
+                json_loads_result = json.loads(json_dumped_dataset_parameters)
+            except TypeError:
```

Comment:
When does a `TypeError` occur?
If the string is invalid JSON you get a `JSONDecodeError`.

Reply:
Up to now, there is no mapper which is dealing with dataset_parameter. I have developed this feature as the first one which is the radarsat2 mapper. If someone want to add sth to this `dataset_parameters` section of metadata. That JSONDecodeError should raise normaly and the developer have to develop sth for that. All the rest of the code is also helping him/her to develop this `dataset_parameters` section in desired mapper. Acually, this `dataset_parameters` section is always loaded based on a previous dump.

Reply:
So, a `TypeError` occurs when the argument to `json.loads()` is not a string, bytes or bytearray.

I still don't understand why you would catch `TypeError` and not `JSONDecodeError`. Both occur when the input is invalid.

Reply:
If JSONDecodeError let it be! but for TypeError, I put TypeError there because:
![image](https://user-images.githubusercontent.com/62345618/105985589-f154dc80-609b-11eb-8f8c-b868a0388fef.png)
and
![image](https://user-images.githubusercontent.com/62345618/105985635-02055280-609c-11eb-895b-bc15397a8a8c.png)
which might happen during the development of other mappers

Reply:
Why would we let it be? It can also happen during the development of other mappers, as you say.

Reply:
Because we want to make sure that this part of metadata comes as a serialized json. serialized json for all the mappers

Reply:
developer should always use `json.dump` there.

Reply:
If you log a message and re-raise the exception as discussed in another comment, you still interrupt the ingestion for the current dataset.
I still don't get why you would want to do that for TypeErrors and not JSONDecodeErrors. In both cases something went wrong in the mapper.

Reply:
yes, you are right. I will delete this try/except and let the two cases raise normal as they happen.